src/sss_client/pam_sss.c

```diff
@@ -594,19 +594,25 @@ static int user_info_expire_warn(pam_handle_t *pamh,
         return PAM_BUF_ERR;
     }
     memcpy(&expire, buf + sizeof(uint32_t), sizeof(uint32_t));
-    if (expire >= DAYSEC) {
-        expire /= DAYSEC;
-        unit = "day(s)";
-    } else if (expire >= HOURSEC) {
-        expire /= HOURSEC;
-        unit = "hour(s)";
-    } else if (expire >= MINSEC) {
-        expire /= MINSEC;
-        unit = "minute(s)";
+    /* expire == 0 indicates the password expired */
+    if (expire != 0) {
+        if (expire >= DAYSEC) {
+            expire /= DAYSEC;
+            unit = "day(s)";
```

Comment:
Thank you for the patch. Overall it looks good.

I wonder if we should use localized string in this (unit = _("day(s)")). I see that it was this way before but since we are touching the code, it might be worth to fix it as well.

What do you think?

Reply:
I missed that. I'll do it.

Reply:
Done now.